src/server/handlers/launchpad.js

```diff
@@ -301,3 +301,57 @@ export const completeSnapAuthorization = async (req, res) => {
     });
   });
 };
+
+export const getSnapBuilds = (req, res) => {
+  const snap_link = req.query.snap_link;
+
+  if (snap_link) {
+    return getLaunchpad().get(snap_link).then(snap => {
+      // TODO: bartaz: currently just limiting to 10 recent builds
+      // we will need to process it properly later for paging/loading more
+      return getLaunchpad().get(snap.builds_collection_link, { start: 0, size: 10 })
+        .then(builds => {
+          return res.status(200).send({
+            status: 'success',
+            payload: {
+              code: 'snap-builds-found',
+              message: builds.entries
+            }
+          });
+        });
+    })
+    .catch(error => {
+      if (error.response) {
+        // At least for the moment, we just wrap the error we get from
+        // Launchpad.
+        return error.response.text().then(text => {
+          logger.info('Launchpad API error:', text);
+          return res.status(error.response.status).send({
+            status: 'error',
+            payload: {
+              code: 'lp-error',
+              message: text
+            }
+          });
+        });
+      } else {
+        return res.status(500).send({
+          status: 'error',
+          payload: {
+            code: 'internal-error',
+            message: error.message
+          }
+        });
+      }
+    });
+  }
+  else {
+    return res.status(404).send({
```

Comment:
I'm generally inclined to put this sort of "is the request well-formed?" check up at the top with an early return, rather than buried down at the bottom of an if/else chain.  This makes it easier to see what condition the error is associated with, and it reduces indentation depth.